### ISSUE TITLE
Multi Component Formly instance?

### ISSUE THREAD

**Issue by zirius666** [ngx-formly/ngx-formly#3546]
In an scenario where you have a very huge larger formly component if you want to divide the ts in components how we can reach this logic in formly ? Since formly is based in ractive forms and forms supports this logic like : https://stackoverflow.com/questions/43270564/dividing-a-form-into-multiple-components-with-validation

i took as base the multi tab example but adding a lot of fields so i wanted for example to divide them at least in separated tab components or so [HERE](https://stackblitz.com/edit/angular-9htzdh?file=src%2Fapp%2Fapp.component.ts)

**Comment by aitboudad** [comment]
see https://formly.dev/docs/guide/formly-field-presets

**Comment by emmanuelOpenForge** [comment]
I dont think this address my question since these is just kind of "minifying" the json but no spliting it

**Comment by aitboudad** [comment]
I think you may need to elaborate more, splitting the form is not clear to me.
Do you means attaching the field dynamically similar to https://formly.dev/docs/examples/advanced/repeating-section/ ?

**Comment by emmanuelOpenForge** [comment]
For example in the [DEMO](https://stackblitz.com/edit/angular-9htzdh?file=src%2Fapp%2Fapp.component.ts) You can see that only for 3 tabs and 10 fields each the file increase it to 200 lines , In my actual project I have 15 tabs and like 20 or more fields with onInit handlers , on change functions Expression Properties , to the file is very very hugh.
- Presets wont work because each field is totally different to other one
- Repeat section Im using but that is onlyt for fieldarrays

So what i want is to "decompose" that big component with tabs into small chunk components , but all of the should point to the same json (But different child properties) Is there a way to accomplish that?

**Comment by aitboudad** [comment]
I see, so instead you can pass a component to `type` property and move all related handlers, extra config to that component:

```ts
fields = [
  ...,
  { type: CustomComponent },
]
```

```ts
import { Component } from '@angular/core';
import { FieldType, FieldTypeConfig } from '@ngx-formly/core';

@Component({
  selector: 'formly-field-input',
  template: `<formly-field *ngFor="let f of field.fieldGroup;" [field]="f"></formly-field>`,
})
export class NameComponent extends FieldType<FieldTypeConfig> {
  // use `defaultOptions` or `prePopulate` function to assign extra properties/handler to the field
}
```

**Comment by emmanuelOpenForge** [comment]
@aitboudad so a ` { type: CustomComponent },` can handle a inner formly form? or should I used default angular reactive forms?

**Comment by aitboudad** [comment]
inner formly form is `formly-field` which handle all child components as well similar to the tab component that you've used in the above demo.

**Comment by emmanuelOpenForge** [comment]
Sorry i got lost because in one side says CustomComponent and in the other NameComponent but you refer that they are the same one , let me test that approach

**Comment by emmanuelOpenForge** [comment]
I can see this [LINK](https://formly.dev/docs/guide/custom-formly-extension/#1-define-the-custom-extension) as the unique resource for prepopulate function and none for defaultOptions , could you please share more info about it? @aitboudad Thanks

**Comment by aitboudad** [comment]
##### `defaultOptions`:
During the initialization we perform a merge of the passed field (`fields` input) and `defaultOptions`  (Example: https://github.com/ngx-formly/ngx-formly/blob/main/src/ui/bootstrap/radio/src/radio.type.ts#L43-L47)

##### `prepopulate`:
it is similar to [custom extenstion](https://formly.dev/docs/guide/custom-formly-extension/#1-define-the-custom-extension)  so the following methods can be used to customize the field:

  - `prePopulate`
  - `onPopulate`
  - `postPopulate`

**Comment by zirius666** [comment]
Thanks I make it work.! 👍

**Comment by ahmedelkadydev** [comment]
> Thanks I make it work.! 👍
Could you please share the working approach cause i have the same issue
